packages/compiler/src/experimental/realm.ts

```diff
@@ -3,6 +3,8 @@ import { Program } from "../core/program.js";
 import { Type } from "../core/types.js";
 import { createTypekit, Typekit } from "./typekit/index.js";
 
+(globalThis as any).realmForType ??= new WeakMap<Type, Realm>();
```

Comment:
should at least use a symbol here to reduce chance of conflict + consider a general purpose `singleton` helper

Reply:
`globalThis[Symbol.for("typespec-realm-for-type")]` or somesuch?

Reply:
for the symbol yeah, but was tghinking maybe we want something like that 

```
class Realm {

  static realmForType = singleton("Realm.realmForType", () => new WeakMap<Type, Realm>());
}

```